core/app/models/spree/order.rb

```diff
@@ -585,28 +606,11 @@ def add_store_credit_payments
       # turned off, and one of the payments fails when the user tries to
       # complete the order, which sends the order back to the 'payment' state.
       authorized_total = payments.pending.sum(:amount)
-
       remaining_total = outstanding_balance - authorized_total
 
-      matching_store_credits = user.store_credits.where(currency: currency)
+      payment_method = Spree::PaymentMethod::StoreCredit.first
 
-      if matching_store_credits.any?
-        payment_method = Spree::PaymentMethod::StoreCredit.first
-        sorter = Spree::Config.store_credit_prioritizer_class.new(matching_store_credits, self)
```

Comment:
IMO it seems pretty obvious that the design of the gem was to have this class be custom configured to handle the scope.  Then the add_ method can be overwritten with a custom opening clause + super.